models/taskRequests.js

```diff
@@ -0,0 +1,173 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const { userState } = require("../constants/userStatus");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+const userStatusModel = require("./userStatus");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  try {
+    const taskRequestsSnapshot = await taskRequestsCollection.get();
+
+    const taskRequests = [];
+
+    const visitedUsers = new Set();
+    const userSnapshot = [];
+    const users = [];
+
+    taskRequestsSnapshot.forEach((taskRequest) => {
+      const taskRequestData = taskRequest.data();
+      const requestors = taskRequestData.requestors;
+
+      requestors.forEach((requestor) => {
+        if (!visitedUsers.has(requestor)) {
+          visitedUsers.add(requestor);
+
+          userSnapshot.push(userModel.fetchUser({ userId: requestor }));
+        }
+      });
+
+      taskRequests.push({ id: taskRequest.id, ...taskRequestData });
+    });
+
+    const usersData = await Promise.all(userSnapshot);
+    usersData.forEach((user) => {
+      users.push(user.user);
+    });
+
+    return {
+      taskRequests,
+      users,
+    };
+  } catch (err) {
+    logger.error("error fetching tasks", err);
+    throw err;
+  }
+};
+
+/**
+ * Creates a task request
+ *
+ * @param taskId { string }: id of task request
+ * @return {Promise<{taskRequest: Object}>}
+ */
+const addOrUpdate = async (taskId, userId) => {
+  try {
+    const taskRequestsSnapshot = await taskRequestsCollection.where("taskId", "==", taskId).limit(1).get();
```

Comment:
do we need to use `limit(1)`, are we going to have multiple taskRequest object for one task? are we not creating a task request object once, and updating that if more people request that task?

Reply:
Removed `limit(1)` in the latest  commit